sfge/src/main/java/com/salesforce/rules/UnusedMethodRule.java

```diff
@@ -109,47 +109,19 @@ private void seekMethodUsages(List<MethodVertex> candidateVertices) {
             // If the method was determined as eligible, track it as such.
             ruleStateTracker.trackEligibleMethod(candidateVertex);
 
-            // Check first for internal usage of the method.
-            InternalCallValidator internalCallValidator =
-                    new InternalCallValidator(candidateVertex, ruleStateTracker);
-            if (internalCallValidator.methodUsedInternally()) {
-                continue;
+            // Depending on the kind of method, we should instantiate a different call validator.
+            BaseMethodCallValidator validator;
```

Comment:
It turns out that it's easier and cleaner to divide the functionality by the type of the method being tested, instead of by the type of reference we're looking for.